### PR TITLE
Extended MockCluster to set a broker down and up again. Also added a …

### PR DESCRIPTION
**Pull request by kkoehler**
Hi,

this PR adds the ability to set a MockCluster down and up again. With this change you're able to test not reachable clusters. A sample is also included.

Kristian

### CODE REVIEW

File: examples/mockcluster_failure_example/mockcluster.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2022 Confluent Inc.
```
[the remark this answers is not in the document]
Comment (kkoehler) [reply examples/mockcluster_failure_example/mockcluster.go]: done in update

File: examples/mockcluster_failure_example/mockcluster.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	}
+
+	fmt.Println("'reset' broker roundtrip duration")
+	err = mockCluster.SetRoundtripDuration(1, 10*time.Millisecond)
+	if err != nil {
+		fmt.Errorf("could not configure roundtrip duration: %v", err)
+		return
+	}
+
+	//see what happens when broker is down
```
[the remark this answers is not in the document]
Comment (kkoehler) [reply examples/mockcluster_failure_example/mockcluster.go]: done in update

File: kafka/mockcluster.go [hunk 3]
```diff
@@ -92,6 +92,26 @@ func (mc *MockCluster) SetRoundtripDuration(brokerID int, duration time.Duration
 	return nil
 }
 
+// SetBrokerDown disconnects the broker and disallows any new connections.
+// This does NOT trigger leader change.
```
[the remark this answers is not in the document]
Comment (kkoehler) [reply kafka/mockcluster.go:96]: done in update

**Comment by milindl** [comment]
Looks good. Made an internal branch, and created a PR using that for CI to run before the merge. #998

**Comment by Sam-sad-Sajid** [comment]
Hi folks. I'm wondering if this feature will be released soon.

File: examples/.gitignore [hunk 4]
```diff
@@ -26,6 +26,7 @@ legacy/consumer_channel_example/consumer_channel_example
 legacy/producer_channel_example/producer_channel_example
 library-version/library-version
 mockcluster_example/mockcluster_example
+mockcluster_example/mockcluster_failure_example
```
Comment (Sam-sad-Sajid) [inline examples/.gitignore:29]: Just out of curiosity: why is this in .gitignore?

Comment (kkoehler) [reply examples/.gitignore:29]: HI. 
Right... both entries for the mock cluster examples seams to be unnecessary. Looks like a copy&paste error.

**Comment by milindl** [comment]
Hi @kkoehler , could you sign the CLA, we'll need that before merging
Thanks

**Comment by kkoehler** [comment]
> Hi @kkoehler , could you sign the CLA, we'll need that before merging Thanks

sorry, i thought that i signed already. done

**Comment by milindl** [comment]
Right, it got reset, I think, for all contributors. Thanks for signing it again.

**Comment by milindl** [comment]
Closing this as I merged the one on the internal branch with some minor changes and after running the CI.

**Pull request closed without merging**